src/lib/dialog/dialog.ts

```diff
@@ -298,12 +308,44 @@ export class MatDialog {
     if (index > -1) {
       this.openDialogs.splice(index, 1);
 
-      // no open dialogs are left, call next on afterAllClosed Subject
+      // If all the dialogs were closed, remove/restore the `aria-hidden`
+      // to a the siblings and emit to the `afterAllClosed` stream.
       if (!this.openDialogs.length) {
+        this._ariaHiddenElements.forEach((previousValue, element) => {
+          if (previousValue) {
+            element.setAttribute('aria-hidden', previousValue);
+          } else {
+            element.removeAttribute('aria-hidden');
+          }
+        });
+
+        this._ariaHiddenElements.clear();
         this._afterAllClosed.next();
       }
     }
   }
+
+  /**
+   * Hides all of the content that isn't an overlay from assistive technology.
+   */
+  private _hideContent() {
+    const overlayContainer = this._overlayContainer.getContainerElement();
+    const siblings = overlayContainer.parentElement!.children;
+
+    for (let i = 0; i < siblings.length; i++) {
+      let sibling = siblings[i];
+
+      if (sibling !== overlayContainer &&
+        sibling !== this._liveAnnouncer.liveElement &&
```

Comment:
Instead of checking this element directly, just check that the element doesn't have `aria-live`? it will accomplish the same thing without needing to change the live announcer API and also safeguards against something that does its own live announcements.